src/v3/plugins/github/graphView.js

```diff
@@ -0,0 +1,274 @@
+    return this._children(repo, GN._Prefix.pull);
+  }
+
+  comments(commentable: GN.CommentableAddress): Iterator<GN.CommentAddress> {
+    return this._children(commentable, GN._Prefix.comment);
+  }
+
+  reviews(pull: GN.PullAddress): Iterator<GN.ReviewAddress> {
+    return this._children(pull, GN._Prefix.review);
+  }
+
+  // TODO figure out how to overload this fn signature
+  parent(child: GN.ChildAddress): GN.ParentAddress {
+    const options = {
+      direction: Direction.OUT,
+      edgePrefix: GE._Prefix.hasParent,
+      nodePrefix: GN._Prefix.base,
+    };
+    const parents: GN.ParentAddress[] = Array.from(
+      this._neighbors(child, options)
+    );
+    if (parents.length !== 1) {
+      throw new Error(
+        `Parent invariant violated for child: ${stringify(child)}`
+      );
+    }
+    return parents[0];
+  }
+
+  authors(content: GN.AuthorableAddress): Iterator<GN.UserlikeAddress> {
+    const options = {
+      direction: Direction.IN,
+      edgePrefix: GE._Prefix.authors,
+      nodePrefix: GN._Prefix.userlike,
+    };
+    return this._neighbors(content, options);
+  }
+
+  _maybeCheckInvariants() {
+    if (this._isCheckingInvariants) {
+      return;
+    }
+    if (process.env.NODE_ENV === "test") {
+      // TODO(perf): If this method becomes really slow, we can disable
+      // it on specific tests wherein we construct large graphs.
+      this.checkInvariants();
+    }
+  }
+
+  checkInvariants() {
+    this._isCheckingInvariants = true;
+    try {
+      this._checkInvariants();
+    } finally {
+      this._isCheckingInvariants = false;
+    }
+  }
+
+  _checkInvariants() {
+    // Check the node type <-> prefix mapping is maintained
+    const nodeTypeToPrefix = {
+      [GN.REPO_TYPE]: GN._Prefix.repo,
+      [GN.ISSUE_TYPE]: GN._Prefix.issue,
+      [GN.PULL_TYPE]: GN._Prefix.pull,
+      [GN.COMMENT_TYPE]: GN._Prefix.comment,
+      [GN.REVIEW_TYPE]: GN._Prefix.review,
+      [GN.USERLIKE_TYPE]: GN._Prefix.userlike,
+    };
+    for (const node of this._graph.nodes({prefix: GN._Prefix.base})) {
+      const structuredNode = GN.fromRaw((node: any));
+      const type = structuredNode.type;
+      const prefix = nodeTypeToPrefix[type];
+      if (prefix == null) {
+        throw new Error(`Invariant: Unexpected node type ${type}`);
+      }
+      if (!NodeAddress.hasPrefix(node, prefix)) {
+        throw new Error(
+          `Invariant: Expected node with type ${type} to have prefix ${NodeAddress.toString(
+            prefix
+          )}`
+        );
+      }
+    }
+
+    // Check that the edge type <-> prefix mapping is maintained.
+    // Check that each edge's src has an expected prefix
+    // Check that each edge's dst has an expected prefix
+    type EdgeInvariant = {|
+      +prefix: GE.RawAddress,
+      +srcPrefixes: NodeAddressT[],
+      +dstPrefixes: NodeAddressT[],
+    |};
+    const edgeTypeToInvariants: {[type: string]: EdgeInvariant} = {
```

Comment:
If you’re inclined, you should be able to specify the type more
precisely using `$ElementType<StructuredAddress, "type">` instead of
`string`. No strong preference.

Reply:
Ack. No action taken.